File: 07_JavaScript/Exercices/exo35/classes/Personnage.js
export default class Personnage {
    nom;
    vie;
    force;

    constructor (nom, vie, force) {
        this.nom = nom;
        this.vie = vie;
        this.force = force;
    }

    attaquer(personnage) {
        personnage.vie -= this.force;
        console.log(`${this.nom} inflige ${this.force} à ${personnage.nom}!`);
    }    

}

File: 19_Java/03_DESIGN_PATTERN/exercices/src/main/java/org/example/exercice03TpBuilderEtAbstractFactory/entity/personnage/Personnage.java
package org.example.exercice03TpBuilderEtAbstractFactory.entity.personnage;

public abstract class Personnage {

    protected String nom;
    protected int force;
    protected int magic;

    protected Personnage(AbstractPersonnageBuilder builder){
        this.nom = builder.nom;
        this.force = builder.force;
        this.magic = builder.magic;
    }
}
